includes/admin/class-wc-rest-payments-accounts-controller.php

```diff
@@ -35,11 +35,39 @@ public function register_routes() {
 	}
 
 	/**
-	 * Get accounts details via API.
+	 * Get account details via API.
 	 *
-	 * @param WP_REST_Request $request Full data about the request.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_account_data( $request ) {
-		return WC_Payments::get_account_service()->get_cached_account_data();
+		$account = WC_Payments::get_account_service()->get_cached_account_data();
+		if ( [] === $account ) {
+			$default_currency = get_woocommerce_currency();
+			$status           = WC_Payments_Account::is_on_boarding_disabled() ? 'ONBOARDING_DISABLED' : 'NOACCOUNT';
```

Comment:
For my learning: how do we disable onboarding for a merchant?

Reply:
We disable it for all =) There is a method in the configuration class on the server-side, which can disable onboarding. My guess is that it's for maintenance purposes.